File: BackEnd/Api/Application/DataTransfer/DockDto.cs
using Api.Domain.ValueObjects;

namespace Api.Application.DataTransfer;

public class DockDto
{
    public required string Code { get; set; }
    public required string Name { get; set; }
    public required string Location { get; set; }
    public required PhysicalCharacteristics PhysicalCharacteristics { get; set; }
    public required List<VesselTypeDto> SupportedVesselTypes { get; set; }
}

public class CreateDockDto
{
    public required string Code { get; set; }
    public required string Name { get; set; }
    public required string Location { get; set; }
    public required PhysicalCharacteristics PhysicalCharacteristics { get; set; }
    public required List<string> SupportedVesselTypes { get; set; }
}

File: FrontEnd/src/model/dto/DockDto.ts
import type { PhysicalCharacteristics } from "../values/PhysicalCharacteristics";

export interface DockDto {
    code: string;
    name: string;
    location: string;
    physicalCharacteristics: PhysicalCharacteristics;
    supportedVesselTypes: string[];
}

export interface DockFilter {
    dockName: string;
    location: string;
    vesselTypeName: string;
}